Teensy/src/gpio/gpio.cpp

```diff
@@ -60,7 +84,60 @@ static int32_t gpio_getGearPosition()
 
 static uint32_t gpio_getSteeringWheelPosition()
 {
-    return 0;
+    #include <Wire.h>
+    #include <Arduino.h>
+   const int MPU = 0x68; //MPU6050 I2C Address
+   float AccX, AccY, AccZ;
+   float GyroX, GyroY, GyroZ;
+   float gyroAngleX, gyroAngleY, gyroAngleZ;
+   float elapsedTime, currentTime, previousTime;
+   float startPosX, startPosY, startPosZ;
+   float currentPosX, currentPosY, currentPosZ;
+   float yaw;
+   float calibrationvalue;
+   int c = 0;
+
+
+    Serial.begin(19200);
+    Wire.begin();
+    Wire.beginTransmission(MPU);
+    Wire.write(0x6B);
+    Wire.write(0x00);
+    Wire.endTransmission(true);
+    delay(20);
+
+
+    // Setting starting position of the sensor //
+
+    startPosX = 0;
+    startPosY = 0;
+    startPosZ = 0;
+
+    // Reading Gyroscope data //
+
+    previousTime = currentTime;
```

Comment:
This won't work unfortunately, because each time this function is called it has no memory of the "old" currentTime value. Main.cpp will be able to track the current and previous time so this can provide this data. However, I think we can seperate this as I assume the steering wheel position might still be useful. 

So could you create an issue for calculating yaw and we will just make this issue about getting these functions complete